packages/langchain/src/orchestration/client.test.ts

```diff
@@ -129,4 +158,93 @@ describe('orchestration service client', () => {
       'Request failed with status code 400'
     );
   }, 1000);
+
+  it('supports streaming responses', async () => {
+    mockStreamInferenceWithResilience();
+
+    const client = new OrchestrationClient(config);
+    const stream = await client.stream([]);
+    let iterations = 0;
+    const maxIterations = 2;
+    let intermediateChunk: AIMessageChunk | undefined;
+
+    for await (const chunk of stream) {
+      iterations++;
+      intermediateChunk = !intermediateChunk
+        ? chunk
+        : intermediateChunk.concat(chunk);
+      if (iterations >= maxIterations) {
+        break;
+      }
+    }
+    expect(intermediateChunk).toBeDefined();
+    expect(intermediateChunk!.content).toBeDefined();
+    expect(intermediateChunk!.content).toEqual(
+      'The SAP Cloud SDK is a comprehensive development toolkit designed to simplify and accelerate the cre'
+    );
+  });
+
+  it('streaming with abort signal', async () => {
+    mockStreamInferenceWithResilience();
+    const client = new OrchestrationClient(config);
+    const controller = new AbortController();
+    const { signal } = controller;
+    const stream = await client.stream([], { signal });
+    const streamPromise = async () => {
+      for await (const _chunk of stream) {
+        controller.abort();
+      }
+    };
+
+    await expect(streamPromise()).rejects.toThrow();
+  }, 1000);
+
+  it('streaming with callbacks', async () => {
+    mockStreamInferenceWithResilience();
+    let tokenCount = 0;
+    const callbackHandler = {
+      handleLLMNewToken: jest.fn().mockImplementation(() => {
+        tokenCount += 1;
+      })
+    };
+    const client = new OrchestrationClient(config, {
+      callbacks: [callbackHandler]
+    });
+    const stream = await client.stream([]);
+    const chunks = [];
+
+    for await (const chunk of stream) {
+      chunks.push(chunk);
+      break;
+    }
+    expect(callbackHandler.handleLLMNewToken).toHaveBeenCalled();
+    const firstCallArgs = callbackHandler.handleLLMNewToken.mock.calls[0];
+    expect(firstCallArgs).toBeDefined();
+    // First chunk content is empty
+    expect(firstCallArgs[0]).toEqual('');
+    // Second argument should be the token indices
+    expect(firstCallArgs[1]).toEqual({ prompt: 0, completion: 0 });
+    expect(tokenCount).toBeGreaterThan(0);
+    expect(chunks.length).toBeGreaterThan(0);
+  });
+
+  it('supports streaming responses with tool calls', async () => {
+    mockStreamInferenceWithResilience(mockResponseStreamToolCalls);
+
+    const client = new OrchestrationClient(config);
+    const stream = await client.stream([]);
+
+    let finalOutput: AIMessageChunk | undefined;
+    for await (const chunk of stream) {
+      finalOutput = !finalOutput ? chunk : finalOutput.concat(chunk);
```

Comment:
```suggestion
      finalOutput = finalOutput ? finalOutput.concat(chunk) : chunk;
```

Reply:
But again, just concat tool call json string together and compare that. We iterate through all chunks anyway. It worths the effort to try to get content from each chunk and make sure it always works for all chunks.

Reply:
Added a snapshot check for `tool_call_chunks`. Both `tool_call_chunks` and `tool_calls` are supposed to have the full function details in the final concatenated chunk.